File: reddio-java/reddio-api/src/main/java/com/reddio/api/v1/rest/OrderMessage.java
package com.reddio.api.v1.rest;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor(staticName = "of")
public class OrderMessage {
    public static final Long DIRECTION_ASK = 0L;
    public static final Long DIRECTION_BID = 1L;

    public static final String STOP_LIMIT_TIME_IN_FORCE_GTC = "GTC";
    public static final String STOP_LIMIT_TIME_IN_FORCE_IOC = "IOC";
    public static final String STOP_LIMIT_TIME_IN_FORCE_FOK = "FOK";

    @JsonProperty("amount")
    private String amount;
    @JsonProperty("amount_buy")
    private String amountBuy;
    @JsonProperty("amount_sell")
    private String amountSell;
    @JsonProperty("token_buy")
    private String tokenBuy;
    @JsonProperty("token_sell")
    private String tokenSell;
    @JsonProperty("base_token")
    private String baseToken;
    @JsonProperty("quote_token")
    private String quoteToken;
    @JsonProperty("vault_id_buy")
    private String vaultIdBuy;
    @JsonProperty("vault_id_sell")
    private String vaultIdSell;
    @JsonProperty("expiration_timestamp")
    private Long expirationTimestamp;
    @JsonProperty("nonce")
    private Long nonce;
    @JsonProperty("signature")
    private Signature signature;
    @JsonProperty("account_id")
    private String accountId;
    @JsonProperty("direction")
    private Long direction;
    @JsonProperty("fee_info")
    private FeeInfo feeInfo;
    @JsonProperty("price")
    private String price;
    @JsonProperty("stark_key")
    private String starkKey;
    @JsonProperty("stop_limit_time_in_force")
    private String stopLimitTimeInForce;
    @JsonProperty("payment")
    private Payment payment;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor(staticName = "of")
    public static class Payment {

        @JsonProperty("pay_info")
        private com.reddio.api.v1.rest.Payment.PayInfo payInfo;
        @JsonProperty("nonce")
        private Long nonce;
        @JsonProperty("signature")
        private Signature signature;
    }
}

File: reddio-csharp/reddio.api/V1/Rest/OrderMessage.cs
using Newtonsoft.Json;

namespace Reddio.Api.V1.Rest
{
    public class OrderMessage
    {
        public const long DIRECTION_ASK = 0;
        public const long DIRECTION_BID = 1;

        public OrderMessage(string amount, string amountBuy, string amountSell, string tokenBuy, string tokenSell,
            string baseToken, string quoteToken, string vaultIdBuy, string vaultIdSell, long expirationTimestamp,
            long nonce, Signature signature, string accountId, long direction, FeeInfo feeInfo, string price,
            string starkKey)
        {
            Amount = amount;
            AmountBuy = amountBuy;
            AmountSell = amountSell;
            TokenBuy = tokenBuy;
            TokenSell = tokenSell;
            BaseToken = baseToken;
            QuoteToken = quoteToken;
            VaultIdBuy = vaultIdBuy;
            VaultIdSell = vaultIdSell;
            ExpirationTimestamp = expirationTimestamp;
            Nonce = nonce;
            Signature = signature;
            AccountId = accountId;
            Direction = direction;
            FeeInfo = feeInfo;
            Price = price;
            StarkKey = starkKey;
        }

        public OrderMessage()
        {
        }

        [JsonProperty("amount")] public string Amount;
        [JsonProperty("amount_buy")] public string AmountBuy;
        [JsonProperty("amount_sell")] public string AmountSell;
        [JsonProperty("token_buy")] public string TokenBuy;
        [JsonProperty("token_sell")] public string TokenSell;
        [JsonProperty("base_token")] public string BaseToken;
        [JsonProperty("quote_token")] public string QuoteToken;
        [JsonProperty("vault_id_buy")] public string VaultIdBuy;
        [JsonProperty("vault_id_sell")] public string VaultIdSell;
        [JsonProperty("expiration_timestamp")] public long ExpirationTimestamp;
        [JsonProperty("nonce")] public long Nonce;
        [JsonProperty("signature")] public Signature Signature;
        [JsonProperty("account_id")] public string AccountId;
        [JsonProperty("direction")] public long Direction;
        [JsonProperty("fee_info")] public FeeInfo FeeInfo;
        [JsonProperty("price")] public string Price;
        [JsonProperty("stark_key")] public string StarkKey;
    }
}